From 0530e5646e1c51d675cb6436c94654311da5041b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sat, 26 Jul 2025 18:29:36 +0200
Subject: [PATCH] Delay calling the color picker

Add a 200 ms delay when calling the color picker portal. This prevents
GtkEntry to steal the focus from the color picker.

Closes: https://gitlab.gnome.org/World/design/contrast/-/issues/35
---
 src/colour_entry.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/colour_entry.rs b/src/colour_entry.rs
index ca50d51..23def60 100644
--- a/src/colour_entry.rs
+++ b/src/colour_entry.rs
@@ -108,6 +108,7 @@ mod imp {
         #[template_callback]
         async fn icon_pressed(&self, pos: gtk::EntryIconPosition, _entry: &gtk::Entry) {
             if pos == gtk::EntryIconPosition::Secondary {
+                gtk::glib::timeout_future(std::time::Duration::from_millis(200)).await;
                 if let Err(err) = self.pick_color().await {
                     tracing::error!("Failed to pick a color {err}");
                 }
-- 
GitLab

